src/NServiceBus.AcceptanceTesting/Support/ScenarioRunner.cs

```diff
@@ -143,17 +146,24 @@ async Task StartEndpoint(ComponentRunner component, CancellationTokenSource cts)
             }
             catch (Exception ex) when (!ex.IsCausedBy(token))
             {
-                cts.Cancel();
-                runDescriptor.ScenarioContext.AddTrace($"Endpoint {component.Name} failed to start.");
+                // signal other endpoints to stop the startup process and immediately abort the outer Task.WhenAll
+                await cts.CancelAsync().ConfigureAwait(false);
+                runDescriptor.ScenarioContext.AddTrace($"Endpoint {component.Name} failed to start: " + ex);
                 throw;
             }
         }
 
-        Task ExecuteWhens(IEnumerable<ComponentRunner> endpoints, CancellationTokenSource cts)
+        async Task ExecuteWhens(IEnumerable<ComponentRunner> endpoints)
         {
-            var whenTimeout = TimeSpan.FromSeconds(60);
-            return endpoints.Select(endpoint => ExecuteWhens(endpoint, cts))
-                .Timebox(whenTimeout, $"Executing given and whens took longer than {whenTimeout.TotalSeconds} seconds.");
+            using var allWhensTimeout = CreateCancellationTokenSource(TimeSpan.FromMinutes(2));
+            // separate CTS as otherwise a failure during endpoint when blocks will cause WaitAsync to throw an OperationCanceledException
+            using var whenTimeout = CreateCancellationTokenSource(TimeSpan.FromSeconds(60));
```

Comment:
```suggestion
            using var allWhensTimeout = CreateCancellationTokenSource(TimeSpan.FromMinutes(1));
            // separate CTS as otherwise a failure during endpoint when blocks will cause WaitAsync to throw an OperationCanceledException
            using var whenTimeout = new CancellationTokenSource();
```